### PR TITLE
setup tests, import fipscrypto package

### PR DESCRIPTION
**Pull request by samantehrani**

### CODE REVIEW

File: tests/email/attachment_test.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# vim: set fileencoding=utf-8 :
```
Comment (samantehrani) [inline tests/email/attachment_test.py]: ignore, all `tests/email/*` are ported from `core`

File: tests/params_test.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import types, pytest
```
Comment (samantehrani) [inline tests/params_test.py]: ignore, ported from `core`

File: tests/protobuf_test.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (samantehrani) [inline tests/protobuf_test.py]: ignore, ported from `core`

File: tests/retry_test.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
+import pytest
```
Comment (samantehrani) [inline tests/retry_test.py]: ignore, ported from `core`

File: pvHelpers/crypto/symm_key/symm_key_v0.py [hunk 5]
```diff
@@ -124,3 +125,10 @@ def decryptBinary(self, cipher):
             raise CryptoException("Invalid header byte {}".format(header_byte))
 
         return message[4:]
+
+    def __eq__(self, other):
+        return self.protocol_version == other.protocol_version and \
+            self._secret == other._secret
```
Comment (hu13) [inline pvHelpers/crypto/symm_key/symm_key_v0.py]: i don't think `other._secret` is accessible?

File: tests/crypto/symm_key_test.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    SYMM_KEY_PROTOCOL_VERSION.V0,
+    SYMM_KEY_PROTOCOL_VERSION.V1
+])
+def test_symmetric_key(protocol_version):
+    k = PVKeyFactory.newSymmKey(protocol_version=protocol_version)
+
+    # encrypt/decrypt
+    plaintext = os.urandom(1024 * 2 + random.randint(0, 1024))
+    ciphers = [k.encryptBinary(plaintext) for _ in range(500)]
+    assert len(ciphers) == len(set(ciphers))
```
Comment (hu13) [inline tests/crypto/symm_key_test.py]: good test.

File: pvHelpers/crypto/asymm_key/asymm_key_v3.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    @params(object, bytes, bytes)
+    def __init__(self, curve25519_pub, p256_pub):
+        super(PublicKeyV3, self).__init__(self.protocol_version)
+        self.curve25519_pub = curve25519_pub
+        self.p256_pub = p256_pub
+
+
+    @params(object, bytes)
+    def sealBinary(self, message):
+        message_with_header = struct.pack(">BBBB", BINARY_BIT | SEAL_BIT, 0x00, 0x00, 0x00) + message
```
[the remark this answers is not in the document]
Comment (samantehrani) [reply pvHelpers/crypto/asymm_key/asymm_key_v3.py]: so they are constants, tied to `protocol_version=0` . so importing them from there

File: pvHelpers/crypto/user_key/user_key_v1.py [hunk 8]
```diff
@@ -61,7 +61,8 @@ def jsonSerialize(self):
         json_serialized = jdumps({
             "private_key": b64_enc_private_key,
             "signing_key": b64_enc_signing_seed,
-            "version": self.key_version
+            "version": self.key_version ,
+            'p': self.protocol_version
```
[the remark this answers is not in the document]
Comment (samantehrani) [reply pvHelpers/crypto/user_key/user_key_v1.py]: well I meant to delete this method all in all. thx for reminding

File: pvHelpers/crypto/key_protocols.py [hunk 9]
```diff
@@ -1,19 +1,27 @@
...
+
     Latest = 1
 
 class ASYMM_KEY_PROTOCOL_VERSION(object):
     V0 = 0
     V1 = 1
     V2 = 2
+    V3 = 3
+
     Latest = 2
```
[the remark this answers is not in the document]
Comment (samantehrani) [reply pvHelpers/crypto/key_protocols.py]: it should be updated at step 4 https://preveil.atlassian.net/wiki/spaces/FEAT/pages/57114626/Server+Client+Deployment+Plan

**Pull request merged**